c_api/concurrentqueue.cpp

```diff
@@ -0,0 +1,55 @@
+#include "concurrentqueue.h"
+#include "../concurrentqueue.h"
+
+typedef moodycamel::ConcurrentQueue<void*> CQType, *CQPtr;
+
+extern "C" {
+
+int moodycamel_cq_create(CQHandle* handle)
+{
+	if (handle == nullptr) {
+		return MOODYCAMEL_BAD_HANDLE;
+	}
+	CQPtr retval = new CQType;
+	*handle = retval;
+	return MOODYCAMEL_OK;
+}
+
+int moodycamel_cq_destroy(CQHandle handle)
+{
+	if (handle == nullptr) {
+		return MOODYCAMEL_BAD_HANDLE;
+	}
+	CQPtr retval = reinterpret_cast<CQPtr>(handle);
```

Comment:
Perhaps this intermediate should be renamed (or even eliminated), since it's not related to the return value of the function